core/shark_turbine/aot/support/procedural/exported_program.py

```diff
@@ -0,0 +1,276 @@
+        elif fake_val is not None:
+            dtype = fake_val.dtype
+        user_output_dtypes.append(dtype)
+
+    return ExportedProgramIntrinsic(
+        entry_func_op, entry_module_call_entry.signature, user_output_dtypes
+    )
+
+
+class _Hooks(FxImporterHooks):
+    def __init__(self, module_builder: ModuleBuilder):
+        self.module_builder = module_builder
+
+    def resolve_literal(self, gni: GraphNodeImporter, literal: Any) -> Optional[Value]:
+        module_builder = self.module_builder
+
+        # We support resolution of tracked reference types. Currently this
+        # only includes Tensors. All others we let the importer do what it
+        # is going to do.
+        if not isinstance(literal, torch.Tensor):
+            return None
+
+        # See if we know about it.
+        mapping = module_builder.global_ref_tracker.track(literal)
+        if mapping.is_empty:
+            # If it is unknown, just let the default importer take it on.
+            return None
+
+        # Already materialized.
+        logger.debug("Resolved defined global for literal %r", mapping)
+        materialized_global: MaterializedGlobal = mapping.value  # type: ignore
+
+        # Emit a global load and conversion.
+        vtensor_type = gni._cc.tensor_to_vtensor_type(literal)
+        loaded_value = util_d.GlobalLoadOp(
+            materialized_global.ir_type, materialized_global.symbol_name
+        ).result
+        converted_value = Operation.create(
+            "torch_c.from_builtin_tensor",
+            results=[vtensor_type],
+            operands=[loaded_value],
+        ).result
+        return converted_value
+
+
+# In https://github.com/llvm/torch-mlir/pull/3046, the FxImporter was
+# extended to accept a "module_op" as an Operation (vs a Module). Switch for
+# compatibility.
+_fx_importer_accepts_module_op = (
+    "module_op" in inspect.getfullargspec(FxImporter).kwonlyargs
+)
+
+
+def _create_fx_importer(module_builder: ModuleBuilder) -> FxImporter:
+    hooks = _Hooks(module_builder)
+    if _fx_importer_accepts_module_op:
+        # New path.
+        return FxImporter(
+            module_op=module_builder.module_op,
+            config_check=False,
+            py_attr_tracker=module_builder.fx_py_attr_tracker,
+            hooks=hooks,
+        )
+    else:
+        # Legacy path.
+        class FakeModule:
+            def __init__(self, op):
+                self._op = module_builder.module_op
+
+            @property
+            def context(self):
+                return self._op.context
+
+            @property
+            def operation(self):
+                return self._op
+
+            @property
+            def body(self):
+                return self._op.regions[0].blocks[0]
+
+        return FxImporter(
+            module=FakeModule(module_builder.module_op),
+            config_check=False,
+            py_attr_tracker=module_builder.fx_py_attr_tracker,
+            hooks=hooks,
+        )
```

Comment:
Yeah, it required a bit of upstream wrangling of https://github.com/llvm/llvm-project/pull/86329 that I didn't want to block on (which is required because of some spidering deps on the Python ExecutionEngine stuff (which is a real mess), which sparse added). Will eventually land that patch/propagate and can drop the fallback.